File: 1636-number-of-substrings-with-only-1s/1636-number-of-substrings-with-only-1s.py
class Solution:
    def numSub(self, s: str) -> int:
        current=0
        count=0
        mod=1e9+7
        for char in s:
            if char=='1':
                current+=1
                count=(current+count)%mod
            else:
                current=0
        
        return int(count)

File: 1636-number-of-substrings-with-only-1s/1636-number-of-substrings-with-only-1s.java
class Solution {
    public int numSub(String s) {
        long count = 0;
        long current = 0;
        long mod = (long)1e9 + 7;

        for (char c : s.toCharArray()) {
            if (c == '1') {
                current++;
                count = (count + current) % mod;
            } else {
                current = 0;
            }
        }
        return (int)count;
    }
}
